### PR TITLE
PE-6472: implement ario sdk integration gar

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
### Overview
Implement the `ario_sdk` package interfacing with the [ario sdk js package](https://github.com/ar-io/ar-io-sdk).

Implement the following API to interface with the SDK:
- getGateways: Retrieves the list of available gateways.
- getIOTokens: Fetches the IO tokens balance for a given address.

### Gateway Selection
Introduces a gateway selector that allows users to change the gateway for data-related requests (excluding uploads and downloads). The selector is available on the login page and the profile card under advanced settings.

This version keeps the focus on the essentials and clearly explains the new functionalities being introduced.
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/0bgq1r3iave68

### CODE REVIEW

File: lib/app_shell.dart [hunk 1]
```diff
@@ -111,64 +112,74 @@ class AppShellState extends State<AppShell> {
...
+                                              : appLocalizationsOf(context)
+                                                  .syncingPleaseWait,
+                                        ),
                                       ),
                                     );
                                   },
                                 );
                               },
                             ),
+                            if (context.read<ConfigService>().flavor !=
```
Comment (thiagocarvalhodev) [inline lib/app_shell.dart:170]: for QA purposes

File: packages/ario_sdk/lib/src/implementations/ario_sdk_web.dart
Comment (karlprieb) [inline packages/ario_sdk/lib/src/implementations/ario_sdk_web.dart:1]: getGateways returns a paginated list. You need to iterate over it using `cursor`

File: packages/ario_sdk/LICENSE [hunk 2]
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
Comment (karlprieb) [inline packages/ario_sdk/LICENSE:1]: Add a LICENSE

File: packages/ario_sdk/web/ario_sdk.min.js
Comment (karlprieb) [inline packages/ario_sdk/web/ario_sdk.min.js:1]: can you add the version to the file name?

File: packages/ario_sdk/lib/src/implementations/ario_sdk_web.dart
Comment (thiagocarvalhodev) [reply packages/ario_sdk/lib/src/implementations/ario_sdk_web.dart:1]: https://github.com/ardriveapp/ardrive-web/pull/1820/commits/f05c2f804f9f50135387ea7195a5b8c64b38b21a

File: packages/ario_sdk/web/src/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    console.log(address);
+  const io = IO.init();
+  // the balance will be returned in mIO as a value
+  const balance = await io
+    .getBalance({
+      address: address,
+    })
+    .then((balance) => new mIOToken(balance).toIO());
+
+  console.log(balance);
```
Comment (karlprieb) [inline packages/ario_sdk/web/src/index.js:45]: remove log

Comment (thiagocarvalhodev) [reply packages/ario_sdk/web/src/index.js]: https://github.com/ardriveapp/ardrive-web/pull/1820/commits/a6e24f3f52e2e6a3eb2d36567583d9ea162b4b33

**Review by karlprieb — APPROVED** [review]

File: lib/authentication/login/views/login_page.dart [hunk 4]
```diff
@@ -155,7 +155,7 @@ class _LoginPageState extends State<LoginPage> {
             logger.d('Login Success, unlocking default profile');
 
             context.read<ProfileCubit>().unlockDefaultProfile(
-                  loginState.user.password,
+                  loginState.user,
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:158]: is this intended?

Comment (karlprieb) [reply lib/authentication/login/views/login_page.dart:158]: I noticed that you're passing everything 👍

**Review by karlprieb — DISMISSED** [review]

**Review by karlprieb — APPROVED** [review]

**Pull request merged**